test/Microsoft.IdentityModel.Tests/IdentityComparer.cs

```diff
@@ -37,108 +37,83 @@
 using Microsoft.IdentityModel.Tokens;
 using Microsoft.IdentityModel.Tokens.Saml;
 using Microsoft.IdentityModel.Xml;
+using System.Collections.ObjectModel;
+using System.Linq;
 
 namespace Microsoft.IdentityModel.Tests
 {
-    public class CompareContext
-    {       
-        List<string> _diffs = new List<string>();
+    public class IdentityComparer
+    {
+        private static readonly Dictionary<string, Func<Object, object, CompareContext, bool>> _equalityDict =
+            new Dictionary<string, Func<Object, object, CompareContext, bool>>()
+        {
+            {typeof(IEnumerable<Claim>).ToString(), AreClaimsEnumsEqual },
+            {typeof(IEnumerable<ClaimsIdentity>).ToString(), AreClaimsIdentitiesEnumsEqual },
+            {typeof(IEnumerable<string>).ToString(), AreStringEnumsEqual },
+            {typeof(IEnumerable<SecurityKey>).ToString(), AreSecurityKeyEnumsEqual },
+            {typeof(Newtonsoft.Json.Linq.JArray).ToString(), AreJArraysEqual },
```

Comment:
What do you mean? I don't think it's possible for ToString() to be called implicitly here?